### ISSUE TITLE
Angular fingerprint calculation with Chebyshev polynomials

### ISSUE THREAD

**Issue by cyoahs** [atomisticnet/aenet#10]
I am reading the source code of fingerprint calculation. In line 817 of 
`src/ext/sfbasis.f90`
>     f = chebyshev_polynomial(cos_ijk, 0.0d0, PI, sfb%a_order)
`chebyshev_polynomial` takes `cos_ijk` as input and `cos_ijk` is from
>     cos_ijk = dot_product(R_ij, R_ik)/(d_ij*d_ik)

The cosine value ranges from -1 to 1, but in line 817, [0.0, PI] is used. Is there any special concerns for this? Besides, why not use radian instead of cosine?

**Comment by nartrith** [comment]
Thank you Yecheng Shao (@cyoahs) for spotting this inconsistency.

The call to `chebyshev_polynomial` in `sfb_angular` assumed that the angle is given in radians between 0 and PI, but it is actually the cosine of the angle which lies between -1 and +1.  This has been fixed in commits 0ec7154b3d1fe01ce7a051ca100328e7c504622a and 77c2c2895a7bfc5cb916dd75e7e6fbe7706075c8 so that now the correct domain is used.

We do not convert the cosine to radians, as this would cause a computational overhead. For the TiO2 example, conversion to radians can slow down the ANN potential evaluation by ~10%!

I ran some quick tests using only angular functions and no radial functions to determine how the domain and the cosine affect the quality of the ANN potential fit.  I did not see any significant variations in the RMSE for the three cases (original, radians, different domain).  So apparently, even though the function domain was not as intended, this did not (significantly) affect the accuracy of the descriptor.

Also thanks to Michael Chen (@michaelstchen) for discussing this bug.
